### ISSUE TITLE
Unexpected RedundantCondition in do/while loop

### ISSUE THREAD

**Issue by TysonAndre** [vimeo/psalm#486]
Expected: no error. The variable is modified later in the loop.
Observed: Two RedundantCondition instances were emitted for the below example

```php
<?php

function main() : void {
    $tried = false;
    do {
        $done = $tried || rand(0,1) > 0; // unexpected RedundantCondition for $tried
        if (!$done) {
            echo "Not done yet\n";
        }
        $tried = true;
    } while (!$done); // unexpected RedundantCondition for $done
}
```

**Comment by muglug** [comment]
After 8747d78c399ba7b4aa1aa2ec87fba6e5361f7ca7 this is the reduced testcase:
```php
do {
    $done = rand(0, 1) > 0;
} while (!$done);
```

**Comment by muglug** [comment]
Thanks – `do` is tricky.
